### ISSUE TITLE
Investigate fixed argument injection to speed up bound/partial functions

### ISSUE THREAD

**Issue by meadsteve** [meadsteve/lagom#68]
At the moment functools.partial is re-applied each time when a wrapped function is invoked. This is useful when getting started but it adds unnecessary slowness.

Maybe something like:

```python
@bind_to_container(container, bind=["thing])
def do_work(thing: AThingIMightNeed):
    thing.do_it()

```

alternative would be something similar to how fastapi does injection:

```python
@bind_to_container_alternative(container)
def do_work(thing: AThingIMightNeed = inject):
    thing.do_it()

```

**Comment by meadsteve** [comment]
Now that I think more about this I prefer the second alternative and I'd like it to be come the default. Potentially having `bind_to_container ` for this new behaviour and `auto_bind_to_container` for the old behaviour.

**Comment by meadsteve** [comment]
implemented by #69
